iphone/Maps/Classes/CustomViews/MapViewControls/TrackRecordingButtonViewController.swift

```diff
@@ -38,31 +39,29 @@ final class TrackRecordingViewController: MWMViewController {
     fatalError("init(coder:) has not been implemented")
   }
 
-  override func viewDidAppear(_ animated: Bool) {
-    super.viewDidAppear(animated)
-    UIView.transition(with: self.view,
-                      duration: kDefaultAnimationDuration,
-                      options: .transitionCrossDissolve,
-                      animations: {
-      self.button.isHidden = false
-    })
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    // async is for smoother appearance
```

Comment:
It looks like a hack, why exactly is it needed/why the appearance is not smooth? Is it because of too much work on main thread? Is it still an issue with avoiding unnecessary calculations?